### PR TITLE
Deprecate AuthConfig pluginAuth in extauth.proto

### PR DESCRIPTION
**Pull request by arianaw66**
# Description

Deprecate AuthConfig `pluginAuth` field in favor of `passThroughAuth`.

## API changes

`pluginAuth` will be removed in a future release, but no changes to the API for now

## Docs changes

Generated docs include a deprecation comment

# Context

See issue description

## Testing steps

No functional changes; nothing to test in this repo

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: changelog/v1.18.0-beta31/deprecate-AuthConfig-pluginAuth.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+changelog:
+- type: FIX
```
Comment (arianaw66) [inline changelog/v1.18.0-beta31/deprecate-AuthConfig-pluginAuth.yaml:2]: I wanted to ensure this change was visible to users, despite it being non-functional. Open to feedback on another type that would make sense here.

Comment (arianaw66) [reply changelog/v1.18.0-beta31/deprecate-AuthConfig-pluginAuth.yaml]: Actually, do OSS users need to be aware of this? Changed to NON_USER_FACING.

**Review by sheidkamp — COMMENTED** [review]
Code changes look good... holding off approval for 2 things:
* make sure there is a docs ticket created to update the docs referring to plugin auth
* just merged a change to split public/internal extauth interfaces that will affect these changes

**Review by sheidkamp — APPROVED** [review]

**Pull request merged**